examples/pytorch/summarization/run_summarization.py

```diff
@@ -512,11 +512,15 @@ def main():
         )
 
     def preprocess_function(examples):
-        # remove pairs where at least one record is None
-
+        warned_about_empty = False
         inputs, targets = [], []
         for i in range(len(examples[text_column])):
-            if examples[text_column][i] and examples[summary_column][i]:
+            if examples[text_column][i]:
+                if not examples[summary_column][i] and not warned_about_empty:
+                    logger.warning(
+                        "Found an empty reference summary. This will be included in the preprocessed dataset."
+                    )
+                    warned_about_empty = True
```

Comment:
Okay I will just keep it simple, hows this?

Reply:
I think `warned_about_empty` should be defined outside `preprocess_function`, as it might get called several times (depending on the processing batch size).

Reply:
Good catch, done.